src/rpcq.lisp

```diff
@@ -573,38 +282,95 @@ We distinguish between the following options for any field type:
+                              (not (getf initargs ,deprecated-by-keyword)))
+                     (warn ,(format nil "~a has been deprecated by ~a."
+                                    slot-keyword deprecated-by-keyword)))))))
+           (slot-accessor-deprecation-warnings (field-spec)
+             (let* ((slot-name (car field-spec))
+                    (slot-keyword (intern (symbol-name slot-name) :keyword))
+                    (field-settings (cdr field-spec)))
+               (alexandria:when-let* ((deprecated-by (getf field-settings :deprecated-by))
+                                      (deprecated-by-keyword (intern (symbol-name deprecated-by) :keyword)))
+                 `((defmethod ,(accessor slot-name) ((obj ,class-name))
+                     (warn ,(format nil "~a has been deprecated by ~a."
+                                    slot-keyword deprecated-by-keyword))
+                     (,(accessor deprecated-by) obj))))))
            (init-spec (json)
              (lambda (slot-name)
                `( ,(intern (symbol-name slot-name) :keyword)
                   (%deserialize (gethash ,(symbol-name slot-name) ,json))))))
-      (alexandria:with-gensyms (obj hash-table)
-        (let ((slot-names (mapcar #'car field-specs)))
-          `(progn
-             (defclass ,class-name ()
-               ,(mapcar #'make-slot-spec field-specs)
-               ,@(when documentation
-                   `((:documentation ,(format-documentation-string documentation)))))
-             
-             (defmethod %serialize ((,obj ,class-name))
-               (with-slots ,slot-names ,obj
-                 (let ((,hash-table (make-hash-table :test #'equal)))
-                   (setf (gethash "_type" ,hash-table) ,(symbol-name class-name))
-                   ,@(loop :for slot :in slot-names
-                           :collect `(setf (gethash ,(symbol-name slot) ,hash-table)
-                                           (%serialize ,slot)))
-                   ,hash-table)))
-
-             (defmethod %deserialize-struct ((type (eql ',class-name)) (payload hash-table))
-               (assert (string= (gethash "_type" payload) ,(symbol-name class-name)))
-               (make-instance ',class-name
-                              ,@(mapcan (init-spec 'payload) slot-names)))
-
-
-             (defmethod print-object ((,obj ,class-name) stream)
-               (print-unreadable-object (,obj stream :type t)
-                 (pprint-indent :block 2)
-                 ,@(loop :for slot :in slot-names
-                         :collect `(format stream
-                                           "~&  ~A -> ~S"
-                                           ,(symbol-name slot)
-                                           (,(accessor slot) ,obj)))))))))))
+    (alexandria:with-gensyms (obj init-args)
+      (let ((slot-names (mapcar #'car field-specs)))
+        `(progn
+           (defclass ,class-name ,parent-name
```

Comment:
Please add `;; comments` above each section of generated code

Reply:
good idea